crypto/BRCryptoWalletManagerClient.c

```diff
@@ -1379,19 +1392,84 @@ cwmGetNonceAsETH (BREthereumClientContext context,
     cryptoWalletManagerGive (cwm);
 }
 
+// MARK: - GEN Callbacks
+
+static void
+cwmGetBlockNumberAsGEN (BRGenericClientContext context,
+                        BRGenericWalletManager manager,
+                        int rid) {
+    BRCryptoWalletManager cwm = cryptoWalletManagerTake (context);
+
+    BRCryptoCWMClientCallbackState callbackState = calloc (1, sizeof(struct BRCryptoCWMClientCallbackStateRecord));
+    callbackState->type = CWM_CALLBACK_TYPE_GEN_GET_BLOCK_NUMBER;
+    callbackState->rid = rid;
+
+    cwm->client.gen.funcGetBlockNumber (cwm->client.context, cwm, callbackState);
+
+    cryptoWalletManagerGive (cwm);
+}
+
+static void
+cwmGetTransactionsAsGEN (BRGenericClientContext context,
+                         BRGenericWalletManager manager,
+                         const char *address,
+                         uint64_t begBlockNumber,
+                         uint64_t endBlockNumber,
+                         int rid) {
+    BRCryptoWalletManager cwm = cryptoWalletManagerTake (context);
+
+    BRCryptoCWMClientCallbackState callbackState = calloc (1, sizeof(struct BRCryptoCWMClientCallbackStateRecord));
+    callbackState->type = CWM_CALLBACK_TYPE_GEN_GET_TRANSACTIONS;
+    callbackState->rid = rid;
+
+    cwm->client.gen.funcGetTransactions (cwm->client.context, cwm, callbackState, address, begBlockNumber, endBlockNumber);
+
+    cryptoWalletManagerGive (cwm);
+}
+
+static void
+cwmSubmitTransactionAsGEN (BRGenericClientContext context,
+                           BRGenericWalletManager manager,
+                           BRGenericWallet wallet,
+                           BRGenericTransfer transfer,
+                           int rid) {
+    BRCryptoWalletManager cwm = cryptoWalletManagerTake (context);
+
+    BRCryptoCWMClientCallbackState callbackState = calloc (1, sizeof(struct BRCryptoCWMClientCallbackStateRecord));
+    callbackState->type = CWM_CALLBACK_TYPE_GEN_SUBMIT_TRANSACTION;
+    callbackState->u.genWithTransaction.wid = wallet;
+    callbackState->u.genWithTransaction.tid = transfer;
+    callbackState->rid = rid;
+
+    BRGenericHash hash = gwmTransferGetHash (cwm->u.gen, transfer);
+    char *hashAsHex = genericHashAsString (hash);
+//    UInt256 txHash = UInt256Reverse (transaction->txHash);
+//    char *hashAsHex = encodeHexCreate (NULL, txHash.u8, sizeof(txHash.u8));
+
+
+    size_t txLength = 0; // BRTransactionSerialize (transaction, NULL, 0);
```

Comment:
Even as a placeholder, this will lead to nasty corruption (on Android). Can we put a TODO and put this as 1 at least?

Reply:
CORE-415